lms/djangoapps/certificates/models.py

```diff
@@ -443,10 +443,10 @@ def save(self, *args, **kwargs):  # pylint: disable=signature-differs
         As well as the COURSE_CERT_CHANGED for any save event.
         """
         super().save(*args, **kwargs)
-        if GeneratedCertificate._meta.proxy:
-            sending_class = GeneratedCertificate.__base__
+        if self._meta.proxy:
+            sending_class = self._meta.proxy_for_model
         else:
-            sending_class = GeneratedCertificate
+            sending_class = GeneratedCertificate.__base__
```

Comment:
```suggestion
            sending_class = self.__class__
```
Now that we are sure `self` is not a proxy the sender can be the same as it was originally.

https://github.com/eduNEXT/edunext-platform/blob/2803f5faff91ccbdc466f0d1a904b9b8a33af881/lms/djangoapps/certificates/models.py#L400